WalletWasabi/WabiSabi/Client/CoinJoin/Manager/CoinJoinManager.cs

```diff
@@ -566,7 +589,7 @@ private async Task HandleCoinJoinFinalizationAsync(CoinJoinTracker finishedCoinJ
 		}
 
 		// If any coins were marked for banning, store them to file
-		if (finishedCoinJoin.BannedCoins.Count != 0)
+		if (CoinPrison is not null && finishedCoinJoin.BannedCoins.Count != 0)
```

Comment:
This is the problem with nulls, we have to be checking for null everywhere. It is much better to have a non-nullable `CoinPrison`